deploy.sh

```diff
@@ -82,10 +82,31 @@ else
     ZIP_FILENAME="$INPUT_ARCHIVE"
 fi
 
+if [ -z "$BUNDLE_TYPE" ]; then
+    # permitted values for BUNDLE_TYPE can be found here: https://docs.aws.amazon.com/codedeploy/latest/userguide/application-revisions-push.html#push-with-cli
+    case "$ZIP_FILENAME" in
+        *.tar)
+            BUNDLE_TYPE=tar
+        *.tar.gz)
+            BUNDLE_TYPE=tgz
+            ;;
+        *)
+            # assume it's a zipfile
+            BUNDLE_TYPE=zip
+            ;;
+    ecas
+fi
 
-if [ "$(unzip -l "$ZIP_FILENAME" | grep -q appspec.yml)" = "0" ]; then
-    echo "::error::$ZIP_FILENAME was not generated properly (missing appspec.yml)."
-    exit 1;
+if [ $BUNDLE_TYPE == 'zip' ]; then
```

Comment:
Yeah interesting.

I should point out that I ran the listing manually, and it also didn't warn about the missing quotes—either because it's not set from untrusted input, or because the default settings of the linter don't warn on this kind of issue.

I agree that it's probably best to quote it—especially if it's sourced from action.yaml.

Reply:
Done

Reply:
(weirdly, the linter also didn't complain about the misspelled `ecas` statement!)